src/foam/nanos/approval/ApprovalDAO.js

```diff
@@ -93,29 +93,25 @@ foam.CLASS({
       javaCode: `
         String daoKey = request.getServerDaoKey() != null && ! SafetyUtil.isEmpty(request.getServerDaoKey()) ? request.getServerDaoKey() : request.getDaoKey();
         DAO dao = (DAO) x.get(daoKey);
-        FObject found = dao.find(request.getObjId()).fclone();
+        FObject obj = dao.find(request.getObjId()).fclone();
 
         DAO userDAO = (DAO) x.get("localUserDAO");
-        Subject subject = new Subject();
-
-        if ( ((ApprovalRequest) request).getCreatedForAgent() != 0 ) {
```

Comment:
@LenoreChen @lchanmann had to revert https://github.com/kgrgreer/foam3/pull/1453 because we found situations where we had to keep realUser in the context. Please let us know if you have any concerns.

Reply:
the code was in a year ago, curious why it's not working just now.

Reply:
it's not that it wasn't working but it was more like we found a scenario where if we used the initiating real user then we might run into a problem because we would lose the privileges that the original real user had. For example, we do permission check against real user in CapabilityAuthService.